src/Drupal/Commands/core/MigrateRunnerCommands.php

```diff
@@ -524,30 +533,74 @@ public function messages(string $migrationId): RowsOfFields
         $idMap = $migration->getIdMap();
         $sourceIdKeys = $this->getSourceIdKeys($idMap);
         $table = [];
-        foreach ($idMap->getMessages() as $row) {
-            unset($row->msgid);
-            $row = (array) $row;
-            // If the message includes useful IDs don't print the hash.
-            if (count($sourceIdKeys) === count(array_intersect_key($sourceIdKeys, $row))) {
-                unset($row['source_ids_hash']);
-            }
-            $sourceIds = $destinationIds = [];
-            foreach ($row as $key => $value) {
-                if (substr($key, 0, 4) === 'src_') {
-                    $sourceIds[$key] = $value;
-                }
-                if (substr($key, 0, 5) === 'dest_') {
-                    $destinationIds[$key] = $value;
+        if (empty($sourceIdKeys)) {
+            // Cannot find one item to extract keys from, no need to process
+            // messages on an empty id map.
+            return new RowsOfFields($table);
+        }
+        if (!empty($options['idlist'])) {
+            // There is not way to retreive a filtered set of messages from an
+            // id map on Drupal Core, right now.
+            // Even if using \Drush\Drupal\Migrate\MigrateIdMapFilter does the
+            // right thing filtering the data on the id map, sadly its
+            // getMessages() method does not take it into account the iterator,
+            // and retrieves data directly, e.g. at sql id map plugin.
+            // On the other side Drupal Core's
+            // \Drupal\migrate\Plugin\MigrateIdMapInterface only allows to
+            // filter by one source ids set, and not by multiple, on
+            // getMessages().
+            // For now, go over known ids passed directly, one at a time a
+            // work-around, at the cost of more queries in the usual sql id map,
+            // which is likely OK for its use, to show only few source ids
+            // messages.
+            foreach (MigrateUtils::parseIdList($options['idlist']) as $source_ids_value) {
```

Comment:
Instead of moving from `$source_ids_value` to `$sourceIdsValue`, I moved to `$sourceIdValues`, since that is used already on other part of the class, ad `executeMigration()`.